code/datums/brain_damage/phobia.dm

```diff
@@ -0,0 +1,124 @@
+/datum/brain_trauma/mild/phobia
+	name = "Phobia"
+	desc = "Patient is unreasonably afraid of something."
+	scan_desc = "phobia"
+	gain_text = ""
+	lose_text = ""
+	var/phobia_type
+	var/next_check = 0
+	var/next_scare = 0
+	var/list/trigger_words
+	//instead of cycling every atom, only cycle the relevant types
+	var/list/trigger_mobs
+	var/list/trigger_objs //also checked in mob equipment
+	var/list/trigger_turfs
+	var/list/trigger_species
+
+/datum/brain_trauma/mild/phobia/New(mob/living/carbon/C, _permanent, specific_type)
+	phobia_type = specific_type
+	if(!phobia_type)
+		phobia_type = pick(SStraumas.phobia_types)
+
+	gain_text = "<span class='warning'>You start finding [phobia_type] very unnerving...</span>"
+	lose_text = "<span class='notice'>You no longer feel afraid of [phobia_type].</span>"
+	scan_desc += " of [phobia_type]"
+	trigger_words = SStraumas.phobia_words[phobia_type]
+	trigger_mobs = SStraumas.phobia_mobs[phobia_type]
+	trigger_objs = SStraumas.phobia_objs[phobia_type]
+	trigger_turfs = SStraumas.phobia_turfs[phobia_type]
+	trigger_species = SStraumas.phobia_species[phobia_type]
+	..()
+
+/datum/brain_trauma/mild/phobia/on_life()
+	..()
+	if(owner.eye_blind)
+		if(phobia_type == "darkness" && prob(25))
+			freak_out() //rip
+		return
+	if(world.time > next_check && world.time > next_scare)
+		next_check = world.time + 50
+		var/list/seen_atoms = view(7, owner)
+
+		if(LAZYLEN(trigger_objs))
+			for(var/obj/O in seen_atoms)
+				if(is_type_in_typecache(O, trigger_objs))
+					freak_out(O)
+					return
+
+		if(LAZYLEN(trigger_turfs))
+			for(var/turf/T in seen_atoms)
+				if(is_type_in_typecache(T, trigger_turfs))
+					freak_out(T)
+					return
+				if((T.get_lumcount(0,1) <= 0.25) && phobia_type == "darkness") //probably snowflake but can't think of a better way without creating a new trauma type entirely
```

Comment:
So, any turf on the screen will trigger the condition?

Reply:
Yes. This is a fear of the concept of darkness, not merely the fear of being blind. Anxiety induced by darkness is frightening.